src/desktop/Workspace.cpp

```diff
@@ -648,6 +653,8 @@ void CWorkspace::rename(const std::string& name) {
     Debug::log(LOG, "CWorkspace::rename: Renaming workspace {} to '{}'", m_iID, name);
     m_szName = name;
 
+    recheckPersistent();
```

Comment:
I meant `CCompositor::ensurePersistentWorkspacesPresent`

Reply:
No, that method does not touch the workspace persistent property in any way. If i call it without calling "recheckPersistent" the workspace stays alive after renaming, even when empty and no persistent rule on its name.

Reply:
well then it should, no?

Reply:
hmm, at a first look it would be extremely inefficient: that method doesn't not know which workspace have changed, so it needs to check all of them, against each rule, and we already know there is only one change, so it becomes o(rules * workspaces) instead of o(rules).

we could mitigate this by passing an optional workspace argument, so the method can restrict the check only to it. do you like this idea?

Reply:
that is fine ye